spdx/cli_tools/convertor.py

```diff
@@ -0,0 +1,197 @@
+@click.option(
+    "--to",
+    "-t",
+    type=click.Choice(["json", "rdf", "yaml", "xml", "tag"], case_sensitive=False),
+    default="undefined",
+)
+@click.option(
+    "--from",
+    "-f",
+    "from_",
+    type=click.Choice(["tag", "rdf"], case_sensitive=False),
+    default="undefined",
+)
+def main(infile, outfile, src, from_, to):
+    """
+    CLI-TOOL for converting a RDF or TAG file to RDF, JSON, YAML, TAG or XML format.
+
+    To use : run 'convertor -f <from_TYPE> <input file> -t <to_TYPE> <output_file>' command on terminal or use ' convertor --infile <input file name> --outfile <output file name> '
+
+    """
+    if infile is None and outfile is None:
+        """
+        when the CLI is of given format:
+        ' convertor -f/--from <type> <input_file> -t/--to <type> <output_file>.
+        """
+        infile = src[0]
+        outfile = src[1]
+        # infile = os.path.splitext(infile)[0]
+        if from_ is not None:
+            infile_path = os.path.splitext(infile)[0]
+            infile = infile_path + "." + from_
+        if to is not None:
+            outfile_path = os.path.splitext(outfile)[0]
+            outfile = outfile_path + "." + to
+
+    elif infile is None and outfile is not None:
+        """
+        ' convertor -f/--from <type> <input_file> --outfile <output_file> '
+        """
+        infile = src[0]
+        if from_ is not None:
+            infile_path = os.path.splitext(infile)[0]
+            infile = infile_path + "." + from_
+
+    elif infile is not None and outfile is None:
+        """
+        ' convertor --infile <input_file> -t/--to <type> <output_file>'
+        """
+        outfile = src[0]
+        if to is not None:
+            outfile_path = os.path.splitext(outfile)[0]
+            outfile = outfile_path + "." + to
+
+    if infile.endswith(".rdf"):
+        infile_format = "rdf"
+    elif infile.endswith(".tag"):
+        infile_format = "tag"
+    else:
+        raise FileTypeError(
+            "INPUT FILETYPE NOT SUPPORTED. (only RDF and TAG format supported)"
+        )
+
+    if outfile.endswith(".rdf"):
+        outfile_format = "rdf"
+    elif outfile.endswith(".tag"):
+        outfile_format = "tag"
+    elif outfile.endswith(".json"):
+        outfile_format = "json"
+    elif outfile.endswith(".xml"):
+        outfile_format = "xml"
+    elif outfile.endswith(".yaml"):
+        outfile_format = "yaml"
+    elif outfile.endswith(".spdx"):
+        outfile_format = "tag"
+    elif outfile.endswith(".rdf.xml"):
+        outfile_format = "rdf"
+    else:
+        raise FileTypeError("OUTFILE FILETYPE NOT SUPPORTED")
+
+    try:
+        func_to_call = infile_format + "_to_" + outfile_format
+        result = globals()[func_to_call](infile, outfile)
+        click.echo(result)
+    except EnvironmentError as e:
+        print(os.strerror(e.errno))
+
+
+def rdf_to_json(infile, outfile):
+    infile = str(infile)
+    outfile = str(outfile)
```

Comment:
input variables can be anything (mostly strings ofc) but as a sanity check I deliberately converted it to string because if somehow there is a file say `12345.spdx` and the user wants to check it, it will break the code.

Reply:
Even if the file name is "12345.spdx", the code won't break.